### PR TITLE
mrc-3558 Configurable message to indicate when no data are available for selected filter combination

### PR DESCRIPTION
**Pull request by Nicolas-Dolan**
To test, open `barchart.html` in the browser and remove all filter options.

image

### CODE REVIEW

File: src/bar/BarChartWithFilters.vue [hunk 1]
```diff
@@ -38,14 +38,19 @@
...
-            <div v-if="!!xAxisLabel" id="chart" class="col-md-9">
+            <div v-if="!!xAxisLabel" id="chart" class="col-md-9" style="position:relative;">
                 <bar-chart-with-errors
                         :chart-data="processedOutputData"
                         :xLabel="xAxisLabel"
                         :yLabel="indicatorLabel"
                         :yFormat="formatValueFunction"
                         :show-errors="showRangesInTooltips"
                         style="width: 100%; height: 100%;"></bar-chart-with-errors>
+                <div v-if="showNoDataMessage" id="noDataMessage" class="px-3 py-2" style="width: 60%; position: absolute; border: 1px solid #adb5bd; bottom: 50%; left: 25%; background: #fff; margin: auto;">
```
Comment (Nicolas-Dolan) [inline src/bar/BarChartWithFilters.vue:49]: Classes added to the repo's stylesheet do not seem to get picked up hence the in line styling.

Comment (EmmaLRussell) [reply src/bar/BarChartWithFilters.vue:49]: They should do I think - could you either have a look into that, or should at least be able to put style into a <style> section in this file?

**Review by EmmaLRussell — COMMENTED** [review]
Looks good! Would be nice to resolve the style issue.

File: src/bar/BarChartWithFilters.vue [hunk 2]
```diff
@@ -210,6 +221,9 @@
             },
             anyFiltersShown() {
                 return this.filterConfig.filters.some((f: Filter) => this.filterIsShown(f.id));
+            },
+            showNoDataMessage() {
+                return this.noDataMessage && this.processedOutputData && this.processedOutputData.datasets && !this.processedOutputData.datasets.length
```
Comment (LekanAnni) [inline src/bar/BarChartWithFilters.vue:226]:
```suggestion
                return this.noDataMessage && this.processedOutputData  && !this.processedOutputData.datasets.length
```

**Review by LekanAnni — APPROVED** [review]
Looking good, see a suggestion below.

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**